guides-markdown/vps-setup.md

```diff
@@ -0,0 +1,129 @@
+# Full Node VPS Setup
+
+```post-author
+Alex Sherbuck
+```
+
+```post-description
+This guide provides installation instructions for a Bcoin full node running on a Digital Ocean of Amazon AWS Virtual Private Server.
+```
+
+## Introduction
+
+Running a full node requires your computer always be online and connected to the Bitcoin network. For most users, a VPS is an elegant 24/7 full node solution.
+
+## Digital Ocean
+
+1. Create an account at digitalocean.com
+2. Choose 'Create' -> 'Droplet' from the account dashboard
+3. Choose 'Ubuntu 16.04' for OS Distribution
+4. Choose '4 GB 2 CPU $20/month' for Droplet Size
+5. Add Block Storage, 500GB $50/month.
+
+### A Note on hardware requirements
+These hardware requirements are for a full node. When run in SPV node, you will not need the additional block storage. To run a full node you must maintain a complete history of all Bitcoin transactions. Without the full history the node is unable to validate transactions. If it cannot validate transactions, it cannot validate blocks or mine transactions to blocks.
+
+SPV Nodes use something called a bloom filter to maintain a smaller set of records. For example, if your Bitcoin address were `1A1zP1eP5QGefi2DMPTfTL5SLmv7DivfNa` your SPV node will ask its network peers for all transactions with an address that starts with `1A1z`. The returned data set will contain extra transactions irrelevant to your wallet. But its storage requirements will be far smaller than a full node. Consequently, SPV Nodes can only verify their own transactions.
+
+6. Choose a data center region.
+7. Add a new ssh key
+
+### Setting Up SSH
+If you already have an ssh key you can copy your pubkey here. If you do not then follow these steps:
+
+`ssh-keygen -t rsa -C "your_email@example.com"`
+
+Accept the defaults.
+
+On Windows your keys are located:
+
+`cd %userprofile%/.ssh`
+
+On Linux/Mac:
+
+`cd ~/.ssh`
+
+Copy the contents of the `id_rsa.pub` file and paste them into Digital Ocean's form.
+
+`ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCsYKEA5LZCDyMF+ZbrPWeVIYso0ZzpIZx9L7R+CGqMPo0mrSlYeeaPbP1btM/Wis4a81EaTM7Y5kkKqZ4XB/LnRWp415XVl5QdtGF2l5tgiy2ootVxEwdrH0lXyGFHEpOwHU6MYdYCd+bgpQwa291Q4bOUJhGxNZ07L/rMtZfWhWL
```

Comment:
this might display better as a code block with triple back ticks rather than a single line.